var/spack/repos/builtin/packages/ginkgo/package.py

```diff
@@ -167,19 +167,28 @@ def setup_build_tests(self):
         with working_dir(smoke_test_path):
             make("install")
 
+        # Above installs tests and data needed for running the tests that
+        # adhere to a relative structure to obtain the data. It is generally
+        # safer for tests to run from the test stage so all files are copied
+        # to the install test root. This allows Spack to automatically copy
+        # them to the test stage prior to running test().
+        install_tree(self.prefix.smoke_tests, self.install_test_root)
```

Comment:
Why not just set
```
@property
def install_test_root(self):
    return self.prefix.smoke_tests
```

Reply:
That seems reasonable in the current context.  My concern would be the potential confusion doing so could cause for people who might use this package as a template for adding their tests.  I supposed comments could address that concern.

IMO it is important to emphasize that files are automatically copied to the test stage directory if they are placed in `self.install_test_root`.

Reply:
Is your suggestion required to get this PR merged?  If so, I'll make it.

Reply:
I think it would be better to have the test method copy the data from its canonical install location to the test stage -- I don't see a good reason to duplicate the data.

Reply:
Thanks. 

IMO. The preferred solution would be for the package's build to install into the install test root so the existing infrastructure copies to the test stage automatically.